lib/src/domain/usecase/implementation/fetch_movies_from_endpoint_usecase.dart

```diff
@@ -0,0 +1,84 @@
+import 'dart:async';
+
+import '/src/domain/usecase/usecase_interface.dart';
+import '../../../core/util/data_state.dart';
+import '../../../core/util/enums.dart';
+import '../../entity/movie.dart';
+import '../../repository/i_movie_repository.dart';
+
+sealed class FetchMoviesFromEndpointUseCase implements IUseCase {
+  Future<DataState<List<Movie>>> _fetchMovies(int page);
+
+  @override
+  Future<DataState> call(dynamic params) {
+    return _fetchMovies(params.page ?? 0);
+  }
+}
+
+class FetchPopularUseCase extends FetchMoviesFromEndpointUseCase {
+  final IMovieRepository repository;
+
+  FetchPopularUseCase({
+    required this.repository,
+  });
+
+  @override
+  Future<DataState<List<Movie>>> _fetchMovies(int page) async {
+    final item = await repository.getMovies(
+      page,
+      MovieEndpoint.popular,
+    );
+    return item;
```

Comment:
You can directly return `await repository...`

Reply:
done